hooks/command

```diff
@@ -0,0 +1,236 @@
+  updates+=("{\"type\":\"${proc_type}\",\"docker_image\":\"$image_id\"}")
+done
+
+payload=$(join_by , "${updates[@]}")
+
+echo "Updating formation"
+
+# Run the equivalant Heroku API as with the cli `heroku container:release`
+# Based on https://devcenter.heroku.com/articles/container-registry-and-runtime#api
+curl -sf -X PATCH "https://api.heroku.com/apps/$app/formation" \
+  -d "{\"updates\":[$payload]}" \
+  -H "Content-Type: application/json" \
+  -H "Accept: application/vnd.heroku+json; version=3.docker-releases" \
+  -H "Authorization: Bearer ${HEROKU_API_KEY:-}" \
+  -o /dev/null
+
+echo ""
+
+version=""
+
+function check_release() {
+  local count=0
+  local streamed=0
+  while true;
+  do
+    count=$((count + 1))
+
+    # Get the most recent release data from the Heroku API
+    # Based on https://github.com/heroku/cli/blob/960d10a4816d8d1215fb8e4b7bf5499a6c0c76c7/packages/container-registry-v5/commands/release.js#L86-L90
+    local latest_release; latest_release=$(curl -sf "https://api.heroku.com/apps/$app/releases" \
+      -H "Content-Type: application/json" \
+      -H "Accept: application/vnd.heroku+json; version=3" \
+      -H "Range: version ..; max=1, order=desc" \
+      -H "Authorization: Bearer ${HEROKU_API_KEY:-}")
+
+    version=$(echo "$latest_release" | jq -r '.[0].version')
+    local current; current=$(echo "$latest_release" | jq -r '.[0].current')
+    local status; status=$(echo "$latest_release" | jq -r '.[0].status')
+    local output_stream_url; output_stream_url=$(echo "$latest_release" | jq -r '.[0].output_stream_url')
+    local summary; summary="[heroku version=$version, current=$current, status=$status]"
+
+    if [ $streamed -eq 0 ] && [ "$status" == "pending" ] && [ "$output_stream_url" != "null" ]; then
+      # Prints the output stream to the current logs
+      # Based on https://github.com/heroku/cli/blob/960d10a4816d8d1215fb8e4b7bf5499a6c0c76c7/packages/container-registry-v5/commands/release.js#L99-L100
+      streamed=1
+      echo "$summary releasing..."
+      status=0
+      # Retry necessary because this will 404 until it's eventually consistent
+      # Based on https://github.com/heroku/cli/blob/960d10a4816d8d1215fb8e4b7bf5499a6c0c76c7/packages/container-registry-v5/lib/streamer.js#L15
+      if ! retry 3 curl -sf "$output_stream_url" -H "Accept: text/event-stream" | grep -e "^data:"; then
+        status=$((status + 1))
+      fi
+      if [ $status -eq 0 ]; then
+        continue;
+      else
+        echo "Failed streaming release output: $output_stream_url"
+        exit 1
+      fi
+    fi
+
+    if [ "$status" == "succeeded" ] && [ "$current" == "true" ]; then
+      echo "$summary success"
+      break;
+    fi
+
+    if [ "$status" == "failed" ]; then
+      echo "$summary failure"
+      exit 1
+    fi
+
+    if [ $count -gt 15 ]; then
+      echo "$summary timeout"
+      exit 1
+    fi
+
+    echo "$summary waiting..."
+    sleep $retry_sleep;
```

Comment:
Should this be configurable?

Reply:
Potentially, version 1.0.0 is fine without it, plus it can be set with env `RETRY_SLEEP` if really needed.